contracts/staking/test/unit_tests/lib_safe_downcast_test.ts

```diff
@@ -0,0 +1,84 @@
+import { blockchainTests, expect, Numberish } from '@0x/contracts-test-utils';
+import { BigNumber, SafeMathRevertErrors } from '@0x/utils';
+import * as _ from 'lodash';
+
+import { artifacts, TestLibSafeDowncastContract } from '../../src/';
+
+blockchainTests('LibSafeDowncast', env => {
+    let testContract: TestLibSafeDowncastContract;
+
+    before(async () => {
+        testContract = await TestLibSafeDowncastContract.deployFrom0xArtifactAsync(
+            artifacts.TestLibSafeDowncast,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+    });
+
+    const MAX_UINT_64 = new BigNumber(2).pow(64).minus(1);
+    const MAX_UINT_96 = new BigNumber(2).pow(96).minus(1);
+    const MAX_UINT_256 = new BigNumber(2).pow(256).minus(1);
+
+    describe('downcastToUint96', async () => {
+        async function verifyCorrectDowncastAsync(n: Numberish): Promise<void> {
+            const actual = await testContract.downcastToUint96.callAsync(new BigNumber(n));
+            expect(actual).to.bignumber.eq(n);
+        }
+        async function verifyDowncastErrorAsync(n: Numberish): Promise<void> {
+            const expectedError = new SafeMathRevertErrors.Uint256DowncastError(
+                SafeMathRevertErrors.DowncastErrorCodes.ValueTooLargeToDowncastToUint96,
+                new BigNumber(n),
+            );
+            const tx = testContract.downcastToUint96.callAsync(new BigNumber(n));
```

Comment:
How about instead of duplicating this call here we just have a `toDownCastError(Numberish): RevertError` function for each describe block and just do `expect(verifyCorrectDowncastAsync(n)).to.revertWith(toDownCastError(n))` inside the failure tests.